zebrad/src/components/tracing/component.rs

```diff
@@ -275,10 +275,16 @@ impl Tracing {
             initial_filter: filter,
             #[cfg(feature = "flamegraph")]
             flamegrapher,
-            _guard,
+            _guard: Some(worker_guard),
         })
     }
 
+    /// Drops guard for worker thread of non-blocking logger,
```

Comment:
I'm not sure we want to shutdown the log writer until the very last moment. See the note about the drop order of this field in the struct definition.

Shutting down here would stop other tasks logging shutdown messages, which could make debugging shutdowns harder.

Reply:
Fixed in https://github.com/ZcashFoundation/zebra/pull/6801/commits/6b553d290759d64255389c43c1dfb6698313b28b

It calls `components().shutdown()` which calls `before_shutdown()` for each component then immediately exits the process.

`TokioComponent` handles interrupts independently, TracingEndpoint and MetricsEndpoint don't have custom Drop impls, and `Tracing` should be okay with `before_shutdown()` + `shutdown()`.